README.md
[![Build Status](https://travis-ci.com/latte-carousel/latte-carousel.svg?branch=master)](https://travis-ci.com/latte-carousel/latte-carousel)
[![npm](https://img.shields.io/npm/v/latte-carousel.svg)](https://npmjs.com/package/latte-carousel)
[![npm](https://img.shields.io/npm/l/latte-carousel.svg)](https://github.com/latte-carousel/latte-carousel/blob/master/LICENSE)

<p align="center">
    <img src="https://lattecarousel.dev/img/colored_small.png" alt="LatteCarousel"/>
</p>

# About

LatteCarousel is a lightweight and responsive carousel without any dependencies.

## Usage

```html
<!-- Package -->
<link rel="stylesheet" href="latte-carousel.min.css" />
<script src="latte-carousel.min.js"></script>

<!-- CDN -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/latte-carousel@1.6.1/dist/latte-carousel.min.css" />
<script src="https://cdn.jsdelivr.net/npm/latte-carousel@1.6.1/dist/latte-carousel.min.js"></script>
```

```html
<div class="latte-carousel" id="carousel">
    <div class="latte-item"></div>
    <div class="latte-item"></div>
    <div class="latte-item"></div>
    <div class="latte-item"></div>
</div>
```

```js
var options = {
    count: 3,
    move: 1,
    touch: true,
    mode: "align",
    buttons: true,
    dots: true,
    rewind: true,
    autoplay: 0,
    animation: 500,
    responsive: {
        "0": { count: 1.5, mode: "free", buttons: false },
        "480": { count: 2.5, mode: "free", buttons: false },
        "768": { count: 3, move: 3, touch: false, dots: false },
        "1440": { count: 4, move: 2, touch: false, dots: false },
    },
};

var carousel = new latte.Carousel("#carousel", options);

// Trigger events
carousel.trigger("next");
carousel.trigger("previous");

carousel.trigger("goto", 0);

carousel.trigger("update");

carousel.trigger("remove");

// Listen events
carousel.on("move", () => {});
carousel.on("moved", () => {});

carousel.off();
```

You can also include latte-carousel using ES6 import and Babel or Webpack.

This package already includes .d.ts files.

## Options

| Name       | Type                 | Default   | Description                            |
| ---------- | -------------------- | --------- | -------------------------------------- |
| count      | Number               | 3         | Number of visible items                |
| move       | Number               | 1         | Number of items to scroll              |
| touch      | Boolean              | false     | Enable touch support                   |
| mode       | String [align, free] | "align"   | Align animation to grid (touch only)   |
| buttons    | Boolean              | true      | Enable carousel buttons                |
| dots       | Boolean              | false     | Enable carousel dots                   |
| rewind     | Boolean              | true      | Enable rewind at the end (or start)    |
| autoplay   | Number               | 0         | Autoplay time in millis (0 to disable) |
| animation  | Number               | 500       | Animation time in millis               |
| responsive | Map<String, Object>  | undefined | Map of options for each breakpoint     |

## Features

-   [x] Responsive options
-   [x] Touch support
-   [x] Stage padding
-   [x] Navigation dots
-   [x] Rewind carousel
-   [x] Autoplay carousel
-   [x] Carousel events

## Browser Support

This library is tested on Google Chrome, but it should work on all major browsers with CSS3 and ES5 support (including IE9+).

## Build

Install dependencies:

```sh
yarn install
```

Build project:

```sh
yarn run build
```

Run example:

```sh
yarn run serve

open http://localhost:8080/example
```

Result files:

-   dist/latte-carousel.min.css
-   dist/latte-carousel.min.js


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.6.1] - 2020-04-24

### Fixed

-   Lint task

## [1.6.0] - 2020-04-24

### Changed

-   Auto-scroll timer reset
-   Updated dependencies

## [1.5.6] - 2019-09-29

### Changed

-   Updated dependencies

## [1.5.5] - 2019-06-23

### Changed

-   Updated dependencies

## [1.5.4] - 2019-03-30

### Fixed

-   Carousel responsive map interface

## [1.5.3] - 2019-03-23

### Fixed

-   Carousel movement not working in Safari (vendor prefixes)

## [1.5.2] - 2019-03-21

### Fixed

-   Export carousel options interface

## [1.5.1] - 2019-03-20

### Added

-   Options documentation

### Fixed

-   Carousel options interface

## [1.5.0] - 2019-03-19

### Added

-   Move count option
-   Default font size for latte-item children
-   Carousel move events
-   Drag free mode

### Changed

-   Updated dependencies

### Fixed

-   Optional configs

## [1.4.1] - 2018-11-01

### Fixed

-   Stage when carousel has fewer items than specified count from [@MatheusBordin](https://github.com/MatheusBordin).

## [1.4.0] - 2018-10-20

### Added

-   New carousel trigger events
-   Validation to avoid creating multiple instances using the same element

### Changed

-   Updated dependencies
-   Webpack CLI
-   Jest config to mock .scss imports

## [1.3.0] - 2018-10-10

### Added

-   Action to cancel tweens

### Changed

-   Updated dependencies

## [1.2.1] - 2018-08-18

### Added

-   CHANGELOG file
-   Setting for animation duration

### Changed

-   Updated dependencies

## [1.2.0] - 2018-08-12

### Fixed

-   TypeScript .d.ts generation

### Changed

-   Animation function

## [1.1.1] - 2018-08-11

### Changed

-   Package keywords

## [1.1.0] - 2018-08-01

### Fixed

-   Nested carousel
-   Carousel cleanup

## [1.0.3] - 2018-07-15

### Changed

-   Carousel creation with element or selector

## [1.0.2] - 2018-07-14

### Changed

-   README file with CDN links and centered logo

## [1.0.1] - 2018-07-09

### Added

-   Unit tests

### Changed

-   Updated dependencies

## [1.0.0] - 2018-07-05

### Added

-   Touch scroll lock

### Changed

-   Updated dependencies

## [0.9.0] - 2018-06-23

### Changed

-   Simplified HTML structure
-   Webpack build

### Fixed

-   Linters

## [0.7.0] - 2018-06-02

### Added

-   Travis CI
-   Issue templates
-   Pagination dots
-   Autoplay

### Fixed

-   Copy global options to responsive options from [@chrsmutti](https://github.com/chrsmutti).

## [0.5.0] - 2018-03-18

### Added

-   Basic carousel support
-   Build tasks

[unreleased]: https://github.com/latte-carousel/latte-carousel/compare/master...develop
[1.5.6]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.5...v1.5.6
[1.5.5]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.4...v1.5.5
[1.5.4]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.3...v1.5.4
[1.5.3]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.2...v1.5.3
[1.5.2]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.1...v1.5.2
[1.5.1]: https://github.com/latte-carousel/latte-carousel/compare/v1.5.0...v1.5.1
[1.5.0]: https://github.com/latte-carousel/latte-carousel/compare/v1.4.1...v1.5.0
[1.4.1]: https://github.com/latte-carousel/latte-carousel/compare/v1.4.0...v1.4.1
[1.4.0]: https://github.com/latte-carousel/latte-carousel/compare/v1.3.0...v1.4.0
[1.3.0]: https://github.com/latte-carousel/latte-carousel/compare/v1.2.1...v1.3.0
[1.2.1]: https://github.com/latte-carousel/latte-carousel/compare/v1.2.0...v1.2.1
[1.2.0]: https://github.com/latte-carousel/latte-carousel/compare/v1.1.1...v1.2.0
[1.1.1]: https://github.com/latte-carousel/latte-carousel/compare/v1.1.0...v1.1.1
[1.1.0]: https://github.com/latte-carousel/latte-carousel/compare/v1.0.3...v1.1.0
[1.0.3]: https://github.com/latte-carousel/latte-carousel/compare/v1.0.2...v1.0.3
[1.0.2]: https://github.com/latte-carousel/latte-carousel/compare/v1.0.1...v1.0.2
[1.0.1]: https://github.com/latte-carousel/latte-carousel/compare/v1.0.0...v1.0.1
[1.0.0]: https://github.com/latte-carousel/latte-carousel/compare/0.9.0...v1.0.0
[0.9.0]: https://github.com/latte-carousel/latte-carousel/compare/0.7.0...0.9.0
[0.7.0]: https://github.com/latte-carousel/latte-carousel/compare/0.5.0...0.7.0
[0.5.0]: https://github.com/latte-carousel/latte-carousel/compare/013746bd4798255465ad248a3423a6fe80d9dab3...0.5.0


example/script.js
window.addEventListener("DOMContentLoaded", function () {
    var options = {
        count: 3,
        move: 1,
        touch: true,
        mode: "align",
        buttons: true,
        dots: true,
        rewind: true,
        autoplay: 0,
        animation: 500,
        responsive: {
            "0": { count: 1.5, mode: "free", buttons: false },
            "480": { count: 2.5, mode: "free", buttons: false },
            "768": { count: 3, move: 3, touch: false, dots: false },
            "1440": { count: 4, move: 2, touch: false, dots: false },
        },
    };

    var carousel = new latte.Carousel(".latte-carousel", options);

    carousel.trigger("next");
    carousel.trigger("previous");
    carousel.trigger("goto", 0);

    carousel.on("move", function () {
        console.log("move");
    });

    carousel.on("moved", function () {
        console.log("moved");
    });

    // carousel.remove();
});
